docs/production/deployment.md

```diff
@@ -327,10 +327,13 @@ HTTP as follows:
 
 #### Configuring Zulip to trust proxies
 
-Before placing Zulip behind a reverse proxy, it needs to be configured to trust
-the client IP addresses that the proxy reports. This is important to have
-accurate IP addresses in server logs, as well as in notification emails which
-are sent to end users.
+Before placing Zulip behind a reverse proxy, it needs to be configured
+to trust the client IP addresses that the proxy reports via the
+`X-Forwarded-For` header. This is important to have accurate IP
+addresses in server logs, as well as in notification emails which are
+sent to end users. Zulip doesn't default to trusting all
+`X-Forwarded-For` headers, because doing so would allow clients to
+spoof any IP address.
```

Comment:
It think this last sentence could use a brief explanation of why this is safe to trust with a reverse proxy (because the reverse proxy takes care of the header, ensuring it's trustable, as far as I understand) - since currently it feels to me  like this creates an immediate question of "hmm, does this mean that clients will become able to spoof IP addresses once this is enabled?"

Reply:
That feels like it's addressed by the immediately following bullet points, but I added a clause that makes it more explicit.